api/shared/condition_types.go

```diff
@@ -0,0 +1,155 @@
+/*
+Copyright 2020 Red Hat
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// +kubebuilder:object:generate:=true
+
+package shared
+
+import (
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Conditions for status in web console
+
+// ConditionDetails used for passing condition information into generic functions
+// e.f. GetDataFromSecret
+type ConditionDetails struct {
+	ConditionNotFoundType   ConditionType
+	ConditionNotFoundReason ConditionReason
+	ConditionErrorType      ConditionType
+	ConditionErrordReason   ConditionReason
+}
+
+// ConditionList - A list of conditions
```

Comment:
would be nice to explain why it should be a list and cannot be a map with Type as its unique key

Reply:
The Condition list is just the standard pattern for operator status conditions.  It is, however, a reasonable question to wonder why it is a list as opposed to a map.  I noticed that the `CoP` util library you shared with us assumes the same approach for Conditions, but it looks like they may be wrapping some sort of map around the list somehow:
```
  // +patchMergeKey=type
  // +patchStrategy=merge
  // +listType=map
  // +listMapKey=type
  Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
```

Reply:
as Andrew said, that's the pattern. the latest condition is the one which is also shown in the web UI, Maybe thats where the list requirement comes from, but I have not checked.